pydicom/tests/test_dataset.py

```diff
@@ -68,6 +68,27 @@ def testAttributeErrorInProperty(self):
         with pytest.raises(AttributeError, match=msg):
             ds.pixel_array
 
+    def test_for_stray_raw_data_element(self):
+        import os
+        dataset = Dataset()
+        dataset.PatientName = 'MacDonald^George'
+        filename = 'test.dcm'
```

Comment:
maybe better use file like object to avoid problems with write permissions, e.g.
```python
    dataset = Dataset()
    dataset.PatientName = 'MacDonald^George'
    fp = DicomBytesIO()
    pydicom.dcmwrite(fp, dataset)
    ....    
```

Reply:
Yes, was planning to do that.  Just wanted to go with his exact test to start with.